### PR TITLE
Accessibility changes to plugin pages

### PR DESCRIPTION
**Pull request by tylerkelly13**
Related Ticket: N/A

Description of Changes:
* Accessibility fixes (mostly)
* Fixed heading levels
* Fixed repeated headings (therefore, repeated ids)
* Fixed most headings marked with bold rather than heading elements
* Restructured a couple poorly structured pages
* Fixed some notes for consistency
* Fixed some tabs to spaces
* Fixed at least one broken example
* Fixed some poorly worded content
* Maybe some other stuff

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: plugins/bbcode.md [hunk 1]
```diff
@@ -57,6 +56,6 @@ tinymce.init({
 });
 ```
 
-## Live example
+## Interactive example
```
Comment (JamesToohey) [inline plugins/bbcode.md:59]: In my latest PR I used "Live" example, it seems this is a bit inconsistent

File: plugins/checklist.md [hunk 2]
```diff
@@ -16,17 +16,30 @@ The **Checklist** plugin helps the user keep track of all required actions by cr
 
 ## Try our Checklist demo
 
+In the {{site.productname}} editor, checklists are presented as lists with small checkboxes on the left hand side of the list items. After the item has been completed, a small tick or check mark is drawn in the box by clicking on it.
```
Comment (JamesToohey) [inline plugins/checklist.md:19]: Do checklists have their checkboxes on the left with RTL? Maybe "next to" is better?

File: plugins/permanentpen.md [hunk 3]
```diff
@@ -115,11 +100,22 @@ The **Permanent Pen Properties** dialog box appears.
 **Result**:
 When the Permanent Pen is enabled, it will apply the new settings to the entered text.
 
-### Using the script
+## Basic setup
 
-Alternatively, the Permanent Pen properties can also be changed by configuring the properties in `tinymce.init`.
+Use the following example shows how to add the Permanent Pen toolbar button:
```
Comment (JamesToohey) [inline plugins/permanentpen.md:105]:
```suggestion
The following example shows how to add the Permanent Pen toolbar button:
```

**Review by EkimChau — APPROVED** [review]

File: plugins/bbcode.md (hunk 1 above)
Comment (tylerkelly13) [reply plugins/bbcode.md:59]: We're switching to this, 'live' is a bit vague.

File: plugins/checklist.md (hunk 2 above)
Comment (tylerkelly13) [reply plugins/checklist.md:19]: Yes, they do. I just moved this info from the bottom of the page, I didn't have time to read it :smile:.

File: plugins/permanentpen.md (hunk 3 above)
Comment (tylerkelly13) [reply plugins/permanentpen.md]: Accepted suggestion: https://github.com/tinymce/tinymce-docs/pull/1704/commits/5c3da3ced4f25d4e42b2787498c6a59d0219e232

File: plugins/checklist.md (hunk 2 above)
Comment (tylerkelly13) [reply plugins/checklist.md]: Fixed by https://github.com/tinymce/tinymce-docs/pull/1704/commits/447f952128e58b8de91df229610c884dbf529341

**Review by JamesToohey — APPROVED** [review]

File: plugins/toc.md [hunk 4]
```diff
@@ -10,11 +10,11 @@ controls: toolbar button, menu item
 {% assign pluginname = "Table of Contents" %}
 {% assign plugincode = "toc" %}
 
-`toc` plugin will generate basic *Table of Contents* and insert it into the editor at the current cursor position. Items for the table will be taken from the headers found in the content.
+The `toc` plugin will generate basic *Table of Contents* and insert it into the editor at the current cursor position. Items for the table will be taken from the headers found in the content.
```
Comment (ashfordneil) [inline plugins/toc.md:13]: Other plugin names have not been `like this` they've just been named directly

**Review by ashfordneil — APPROVED** [review]

File: plugins/image.md [hunk 5]
```diff
@@ -104,7 +104,7 @@ tinymce.init({
 });
 ```
 
-**Example of JSON URL with images**
+#### Example of an external script that returns an json array of images
```
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply plugins/image.md:107]:
```suggestion
#### Example of an external script that returns an JSON array of images
```

File: plugins/link.md [hunk 6]
```diff
@@ -165,7 +165,7 @@ tinymce.init({
 });
 ```
 
-**Example of a JSON URL that returns an array of link items**
+#### Example of an external script that returns a json array of link items
```
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply plugins/link.md:168]:
```suggestion
#### Example of an external script that returns a JSON array of link items
```

File: plugins/toc.md (hunk 4 above)
Comment (tylerkelly13) [reply plugins/toc.md:13]: yeah, I can't fix all the docs in one day :) but yes.

Comment (tylerkelly13) [reply plugins/toc.md:13]:
```suggestion
The Table of Contents plugin will generate basic *Table of Contents* and insert it into the editor at the current cursor position. Items for the table will be taken from the headers found in the content.
```

File: plugins/importcss.md [hunk 7]
```diff
@@ -112,7 +114,7 @@ tinymce.init({
 
 This option enables group matching selectors into submenus in the `Formats` menu dropdown. You can use a `string`, `regexp` or a `function` to filter selectors. You can also omit the filter to get all non-matching ones into a specific group. You can also specify a group specific `selector_converter` so that formats for a specific group are produced by that converter.
 
-**Example of usage with string filter**
+#### Example of `importcss_groups` using a string filter
```
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply plugins/importcss.md:117]:
```suggestion
#### Example using a string filter with `importcss_groups` 
```

**Review by ashfordneil — APPROVED** [review]

**Pull request merged**